src/controllers/midi/midimessage.h

```diff
@@ -178,42 +179,52 @@ struct MidiKey {
     };
 };
 
+template<class T>
 struct MidiInputMapping {
-    MidiInputMapping() {
-    }
+    static_assert(std::is_base_of<InputHandlerExecutable, T>::value);
 
+    MidiInputMapping()
+            : control(T()) {
+    }
     MidiInputMapping(MidiKey key, MidiOptions options)
             : key(key),
-              options(options) {
+              options(options),
+              control(ConfigKey()) {
     }
 
-    MidiInputMapping(MidiKey key, MidiOptions options, const ConfigKey& control)
+    MidiInputMapping(MidiKey key, MidiOptions options, const T& control)
             : key(key),
               options(options),
               control(control) {
     }
 
     MidiInputMapping(MidiKey key,
             MidiOptions options,
-            const ConfigKey& control,
+            const T& control,
             const QString& description)
             : key(key),
               options(options),
               control(control),
               description(description) {
     }
 
+    MidiInputMapping(MidiKey key, const QJSValue& control)
+            : key(key),
+              control(control) {
+    }
+
     bool operator==(const MidiInputMapping& other) const {
         return key == other.key && options == other.options &&
                 control == other.control && description == other.description;
     }
 
     MidiKey key;
     MidiOptions options;
-    ConfigKey control;
+    T control;
```

Comment:
I'm afraid we had a misunderstanding with my suggestion here. All I was suggesting was to change this to 
```suggestion
    std::variant<ConfigKey, QJSValue> control;
```

And then dispatch accordingly using `std::visit`. Does that make sense?

Reply:
Quick question here: what's the difference between [this function](https://github.com/mixxxdj/mixxx/blob/main/src/controllers/midi/midicontroller.cpp#L263) and [this one](https://github.com/mixxxdj/mixxx/blob/main/src/controllers/midi/midicontroller.cpp#L527) ?

Reply:
one is for the standard 3-byte midi, the other one (with the QByteArray) is intended for arbitrary binary data such as sysex.